### PR TITLE
delete email feat

### LINKED ISSUE #87: It should be possible to delete an email from the UI
The backend code to delete an email is already there but in the UI currently you can't delete a single email, only a campaign.

### PR DESCRIPTION
**Pull request by icaraps**
Fix #87 

Test URLs:
- Before: https://main--eds-self-service-website--headwire-edge-delivery.aem.live/
- After: https://de--eds-self-service-website--headwire-edge-delivery.aem.live/

### CODE REVIEW

File: blocks/site-details/renderCampaignsOverview.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ export default async function renderCampaignsOverview({
 
   const setCampaignLink = (action, campaign) => {
     if (projectDetails.darkAlleyProject) {
-      action.href = `https://da.live/#/${daProjectRepo}/${siteSlug}/${campaign}`;
+      action.href = `https://da.live/#/${daProjectRepo}/${siteSlug}/emails/${campaign}`;
```
Comment (icaraps) [inline blocks/site-details/renderCampaignsOverview.js:47]: da link fix

File: blocks/site-details/renderSiteOverview.js [hunk 2]
```diff
@@ -122,8 +122,9 @@ export default async function renderSiteOverview({ container, nav, renderOptions
   // MARK: delete site
   container.querySelector('.delete').onclick = async () => {
     window?.zaraz?.track('click site delete');
+    const block = container.closest('.site-details.block');
```
Comment (icaraps) [inline blocks/site-details/renderSiteOverview.js:125]: loading status fix

File: blocks/site-details/renderSitePages.js [hunk 3]
```diff
@@ -21,6 +21,11 @@ export function renderTable({
     const tableRow = document.createElement('tr');
 
     if (type === 'emails') {
+      const split = item.path.split('/');
+      const isDeletable = item.path.startsWith('/emails/') && split.length === 4;
```
Comment (icaraps) [inline blocks/site-details/renderSitePages.js:25]: which makes /newsletter not deletable

**Comment by DriftingSands** [comment]
- Email docs are not deleted if they do not match the slug (mine had a space): https://drive.google.com/drive/folders/1Wm2TGJZFEDuyY_gkshtxXVcYkhrz7b5l
- DB data for email (under `/emails/PROJECT_NAME/EMAIL_ID`) is not deleted when deleting campaign. I think it might be better to store them under the campaigns stuff anyway? i.e. `/campaigns/PROJECT_NAME/CAMPAIGN_NAME/emails/EMAIL_ID` It would make things simpler I think.
- After deleting my campaign, the emails were still in the all-emails list. After refreshing the list breaks and no longer renders. Error: 
image
- Analytics still exist for deleted emails. How should we handle this?




> DB data for email (under `/emails/PROJECT_NAME/EMAIL_ID`) is not deleted when deleting campaign.
I think this has been happening for a while. We have many test project names in that list. They should have been deleted with campaign & with the project, but they weren't

>  After deleting my campaign, the emails were still in the all-emails list. After refreshing the list breaks and no longer renders.
This was due to the url not updating once deleted... It broke because it was trying to load a campaign that doens't exist. Clicking the all-emails button would not change the url either. I.e. I was not redirected to `/emails` after deleting campaign. Refreshing on `/emails/new-campaign` caused issues.

File: blocks/site-details/renderCampaignsAnalytics.js [hunk 4]
```diff
@@ -246,4 +247,9 @@ export default async function renderCampaignsAnalytics({
   });
 
   calculateCampaignStats(window.location.pathname.startsWith(`${pathname}/campaign-analytics/`));
+
+  if (!container.querySelector('.campaign-list .selected')) {
```
Comment (icaraps) [inline blocks/site-details/renderCampaignsAnalytics.js:251]: fixes the issue if no campaign matches the path in campaign analytics

File: blocks/site-details/renderCampaignsOverview.js [hunk 5]
```diff
@@ -130,6 +133,11 @@ export default async function renderCampaignsOverview({
     });
   });
 
+  if (!container.querySelector('.campaign-list .selected')) {
```
Comment (icaraps) [inline blocks/site-details/renderCampaignsOverview.js:136]: fixes the issue if no campaign matches the path in campaign overview

File: blocks/site-details/renderCampaignsOverview.js [hunk 6]
```diff
@@ -375,15 +383,16 @@ export default async function renderCampaignsOverview({
...
+
+          container.querySelector(`.campaign-${campaignSlug}`).remove();
+          container.querySelectorAll(emailsToDelete.join(',')).forEach((el) => {
+            el.remove();
+          });
+
           container.querySelector(`.campaign-list[data-type="emails"] li[data-campaign="${campaignSlug}"]`).remove();
           container.querySelector('.campaign-list[data-type="emails"] a').click();
 
-          const li = container.querySelector(`.campaign-list[data-type="analytics"] li[data-campaign="${campaignSlug}"]`);
```
Comment (icaraps) [inline blocks/site-details/renderCampaignsOverview.js:381]: This is no longer needed as analytics tab is not in the DOM anymore

File: blocks/site-details/renderSitePages.js [hunk 7]
```diff
@@ -50,11 +49,17 @@ export function renderTable({
...
-              headers: { Authorization: `Bearer ${token}` },
+              headers: { Authorization: `Bearer ${token}`, 'content-type': 'application/json' },
+              body: JSON.stringify({
+                id: deleteEmail.dataset.id,
+              }),
             }).catch(() => null);
 
             if (deleteReq?.ok) {
-              deleteEmail.closest('tr').remove();
+              const emails = table.closest('.campaign-container').querySelectorAll(`.campaign tr[data-path="/emails/${campaign}/${email}"]`);
```
Comment (icaraps) [inline blocks/site-details/renderSitePages.js:59]: remove email from all email tables

File: blocks/site-details/renderSitePages.js [hunk 8]
```diff
@@ -50,11 +49,17 @@ export function renderTable({
             deleteEmail.classList.add('loading');
             const deleteReq = await fetch(`${SCRIPT_API}/campaigns/${projectDetails.projectSlug}/${campaign}/${email}`, {
               method: 'DELETE',
-              headers: { Authorization: `Bearer ${token}` },
+              headers: { Authorization: `Bearer ${token}`, 'content-type': 'application/json' },
+              body: JSON.stringify({
+                id: deleteEmail.dataset.id,
```
Comment (icaraps) [inline blocks/site-details/renderSitePages.js:54]: add google drive file id for easy deletion

File: blocks/site-details/renderCampaignsOverview.js [hunk 9]
```diff
@@ -254,20 +262,20 @@ export default async function renderCampaignsOverview({
         campaignContainer.insertAdjacentHTML('beforeend', `
             <div class="campaign campaign-${newCampaign.slug}" hidden>
               <div class="cards">
-                <div>
+                <div class="box">
```
Comment (icaraps) [inline blocks/site-details/renderCampaignsOverview.js:265]: missing box classes after .cards styles refactoring

**Comment by DriftingSands** [comment]
- When creating a new campaign, I am automatically put in the newly created tab for it. But the URL does not update.
- Deleting a campaign with no emails causes an error. See image

image

**Pull request merged**